packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:quiver_hashcode/hashcode.dart';
+
+enum SignInOption { standard, games }
+
+class GoogleSignInUserData {
+  GoogleSignInUserData(
+      {this.displayName, this.email, this.id, this.photoUrl, this.idToken});
+  String displayName;
+  String email;
+  String id;
+  String photoUrl;
+  String idToken;
+
+  @override
+  int get hashCode =>
+      hashObjects(<String>[displayName, email, id, photoUrl, idToken]);
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other)) return true;
+    if (other is! GoogleSignInUserData) return false;
+    return other.displayName == displayName &&
```

Comment:
Prefer to cast the `other` into a `GoogleSignInUserData` so calling `other.email` doesn't have dynamic calls. See https://github.com/flutter/flutter/blob/ab06d033ca210964c541f09c53803590aa95e865/packages/flutter/lib/src/foundation/key.dart#L70 for an example.

Same in the other `operator ==` overrides.

Reply:
It was done the right way as well here in the operator == on google_sign_in.dart. Cool catch, I remember reading about this recently in the chat!